crates/bevy_asset/src/assets.rs

```diff
@@ -286,7 +286,7 @@ pub trait AddAsset {
     /// This enables reflection code to access assets. For detailed information, see the docs on [`ReflectAsset`] and [`ReflectHandle`].
     fn register_asset_reflect<T>(&mut self) -> &mut Self
     where
-        T: Asset + Reflect + FromReflect + GetTypeRegistration;
+        T: Asset + PartialReflect + FromReflect + GetTypeRegistration;
```

Comment:
Do we want to allow proxy-like assets? My feeling is that we do, but just want to check before we enable this by changing the bound to `PartialReflect`.

Reply:
this is just a mistake on my end! this is already bound `T: Reflect` since `FromReflect: Reflect`!

Reply:
Oh lol I didn't see that either